include/gpio_pin.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "gpio_wrapper.hpp"
+#include "stdint.h"
+
+namespace edge {
+namespace drivers {
+
+enum GPIODirection { IN, OUT };
+
+class GPIOPin {
```

Comment:
This is supposed to be bottom half right? just confirming

Reply:
sure